internal/database/models/models.go

```diff
@@ -2,24 +2,30 @@ package models
 
 import (
 	"gorm.io/gorm"
+	"time"
 )
 
 type Author struct {
-	gorm.Model
-	FirstName     string  `gorm:"not null" json:"first_name"`
-	LastName      string  `json:"last_name"`
-	Bio           string  `gorm:"type:text" json:"bio"`
-	OriginCountry string  `gorm:"not null" json:"origin_country"`
-	CurrentCity   string  `gorm:"not null" json:"current_city"`
-	Latitude      float64 `gorm:"not null" json:"latitude"`
-	Longitude     float64 `gorm:"not null" json:"longitude"`
-	Stories       []Story `gorm:"foreignKey:AuthorID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" json:"stories,omitempty"`
+	ID        		uint	`gorm:"autoIncrement:true"`		
+	CreatedAt 		time.Time
+	UpdatedAt 		time.Time
+	DeletedAt 		gorm.DeletedAt `gorm:"index"`
+	FirstName     	string  `gorm:"primaryKey;not null" json:"first_name"`
+	LastName      	string  `gorm:"primaryKey" json:"last_name"`
```

Comment:
Multiple primary keys doesn't make sense, there should only be one (hence *primary* key).

I don't think it should be the database concern to check for uniqueness, that is an application concern (as fields and field types could easily change with a migration).

When we are create new authors, we can query and check for the existing authors. If one of the existing authors are good enough, we can select one of them, otherwise we create a new one. This also means we don't lose the possibility of people having the same firstname/lastname/origin.

Reply:
yep, so we decided with @dhillondeep that the uniqueness of the author will be determined by the combo of their first/last/country, which is why we decided to make the three fields into a composite primary key. (we don't want the possibility of authors have the same first/last/country. for example, if author1 and author2 have the same firstname/lastname/country fields, we assume they're the same author. and i believe we have a separate issue to make `current_city` part of the `stories` model)

> I don't think it should be the database concern to check for uniqueness

imo, it's a good idea to make sure our primary keys are unique at the lowest level to ensure data consistency. right now (with this pr) we are just adding constraints for our db, i.e. data validation (not business logic). i don't think creating a composite primary key is equivalent to, for example, checking that the `origin_country` is indeed an actual country...as for adding a separate layer of validation by querying - well, we'd still want to filter based on first&last&country, so we'd prob want to have an index on those fields anyway...

did i understand your concern correctly?